x-pack/metricbeat/module/aws/sqs/sqs.go

```diff
@@ -107,16 +122,31 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 		}
 
 		// Create Cloudwatch Events for SQS
-		event, err := createSQSEvents(metricDataResults, metricsetName, regionName, schemaRequestFields)
-		if err != nil {
-			m.logger.Error(err.Error())
-			event.Error = err
+		for _, queueURL := range queueURLs {
```

Comment:
In general I would advice to not have `for` loops inside `for` loops especially if they contain quite a bit of code and instead use functions/methods. This gives also the opportunity to give some nice names to the functions and potentially have docs there. This will also help with testability.

Reply:
Good point, I don't know in this case if helped or not :-) I moved the for loop into `createSQSEvents`. Thanks!